### PR TITLE
Adds a UI for gas mixers

### PR DESCRIPTION
**Pull request by ike709**
## About the PR 
See #4203

Adds a UI for mixers.

**Screenshots**

image

**Changelog**

:cl:
- add: Gas mixers now have a UI
- fix: Gas filters and pumps now need to be anchored to open their UIs

### CODE REVIEW

**Review by PaulRitter — CHANGES_REQUESTED** [review]
lgtm, just two things:
- do not open the window if the mixer isnt anchored
- I'd rather have "{Dir} Node" than the numbers, seems more clear

**Comment by ike709** [comment]
CBA to do node directions but I did improve it. It's more immediately clear than using dirs anyways.

File: Resources/Locale/en-US/components/gas-mixer-component.ftl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+comp-gas-mixer-ui-mixer-status = Status:
+comp-gas-mixer-ui-status-enabled = On
+comp-gas-mixer-ui-status-disabled = Off
+
+comp-gas-mixer-ui-mixer-output-pressure = Output Pressure (kPa):
+
+comp-gas-mixer-ui-mixer-node-primary = Primary Node:
+comp-gas-mixer-ui-mixer-node-side = Side Node:
```
Comment (Efruit) [inline Resources/Locale/en-US/components/gas-mixer-component.ftl:8]: While technically correct, I don't think "Node" is the best thing to call them.

```suggestion
comp-gas-mixer-ui-mixer-node-primary = Primary Port:
comp-gas-mixer-ui-mixer-node-side = Side Port:
```

**Review by PaulRitter — CHANGES_REQUESTED** [review]
please update your variables in code too to be consistent with "Primary/Side Node"

fix the nitpicks and i merge

File: Resources/Locale/en-US/components/gas-pump-component.ftl [hunk 2]
```diff
@@ -8,3 +8,5 @@ comp-gas-pump-ui-pump-set-max = Max
 comp-gas-pump-ui-pump-output-pressure = Output Pressure (kPa):
 
 comp-gas-pump-ui-pump-transfer-rate = Transfer Rate (L/s):
+
+comp-gas-pump-ui-needs-anchor = Anchor the pump first!
```
[the remark this answers is not in the document]
Comment (ike709) [reply Resources/Locale/en-US/components/gas-pump-component.ftl:12]:
```suggestion
comp-gas-pump-ui-needs-anchor = Anchor it first!
```

File: Resources/Locale/en-US/components/gas-mixer-component.ftl [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+comp-gas-mixer-ui-mixer-output-pressure = Output Pressure (kPa):
+
+comp-gas-mixer-ui-mixer-node-primary = Primary Port:
+comp-gas-mixer-ui-mixer-node-side = Side Port:
+
+comp-gas-mixer-ui-mixer-set = Set
+comp-gas-mixer-ui-mixer-max = Max
+
+comp-gas-mixer-ui-needs-anchor = Anchor the mixer first!
```
[the remark this answers is not in the document]
Comment (ike709) [reply Resources/Locale/en-US/components/gas-mixer-component.ftl:13]:
```suggestion
comp-gas-mixer-ui-needs-anchor = Anchor it first!
```

File: Resources/Locale/en-US/components/gas-filter-component.ftl [hunk 4]
```diff
@@ -8,3 +8,5 @@ comp-gas-filter-ui-filter-set-rate = Set
 comp-gas-filter-ui-filter-gas-current = Currently Filtering:
 comp-gas-filter-ui-filter-gas-select = Select a gas to filter out:
 comp-gas-filter-ui-filter-gas-confirm = Set Gas
+
+comp-gas-filter-ui-needs-anchor = Anchor the filter first!
```
[the remark this answers is not in the document]
Comment (ike709) [reply Resources/Locale/en-US/components/gas-filter-component.ftl:12]:
```suggestion
comp-gas-filter-ui-needs-anchor = Anchor it first!
```

**Comment by ike709** [comment]
The variable names are consistent with the actual nodes in the code being node one & two. Primary/side is just to make it clearer for players.

**Comment by ike709** [comment]
Linter failure is unrelated to the PR.

**Pull request merged**